config/samples/bpfman.io_v1alpha1_bpfapplication.yaml

```
@@ -9,50 +9,113 @@ spec:
-        priority: 55
-        direction: ingress
-    - type: TCX
+        attach_points:
+          - interfaceselector:
+              primarynodeinterface: true
+            priority: 55
+            direction: ingress
+    - bpffunctionname: tcx_next_test
+      type: TCX
       tcx:
-        bpffunctionname: tcx_stats
-        interfaceselector:
-          primarynodeinterface: true
-        priority: 500
-        direction: ingress
-    - type: Uprobe
+        attach_points:
+          - interfaceselector:
+              primarynodeinterface: true
+            priority: 500
+            direction: ingress
+          - interfaceselector:
+              interfaces:
+                - eth0
+            priority: 100
+            direction: egress
+            containers:
+              namespace: bpfman
+              pods:
+                matchLabels:
+                  name: bpfman-daemon
+              containernames:
+                - bpfman
+                - bpfman-agent
+    - bpffunctionname: uprobe_test
+      type: Uprobe
+      uprobe:
+        attach_points:
+          - func_name: malloc
+            target: libc
+            retprobe: false
+            containers:
+              namespace: bpfman
+              pods:
+                matchLabels:
+                  name: bpfman-daemon
+              containernames:
+                - bpfman
+                - bpfman-agent
+    - bpffunctionname: uretprobe_test
+      type: Uprobe
       uprobe:
-        bpffunctionname: uprobe_counter
-        func_name: malloc
-        target: libc
-        retprobe: false
-        containers:
-          namespace: bpfman
-          pods:
-            matchLabels:
-              name: bpfman-daemon
-          containernames:
-            - bpfman
-            - bpfman-agent
-    - type: XDP
+        attach_points:
+          - func_name: malloc
+            target: libc
+            retprobe: true
+            containers:
+              namespace: bpfman
+              pods:
+                matchLabels:
+                  name: bpfman-daemon
+              containernames:
+                - bpfman
+                - bpfman-agent
+    - bpffunctionname: xdp_pass_test
+      type: XDP
       xdp:
-        bpffunctionname: xdp_stats
-        interfaceselector:
-          primarynodeinterface: true
-        priority: 55
+        attach_points:
+          - interfaceselector:
+              primarynodeinterface: true
+            priority: 55
+          - interfaceselector:
+              interfaces:
+                - eth0
+            priority: 100
+            containers:
+              namespace: bpfman
+              pods:
+                matchLabels:
+                  name: bpfman-daemon
+              containernames:
+                - bpfman
+                - bpfman-agent
+    - bpffunctionname: fentry_test
+      type: Fentry
+      fentry:
+        function_name: do_unlinkat
+        attach: true    
```

```suggestion
    - function: fentry_test
      fentry:
        links:
          - function_name: do_unlinkat
```

This shouldn't deviate from the pattern in the other program types.
The fact we need the fn_name for `load()` and we can only attach it to one place are implementation details we shouldn't be exposing in the API.

---

I've changed the pattern.